synapse/rest/media/v1/_base.py

```diff
@@ -96,7 +110,13 @@ def add_file_headers(request, media_type, file_size, upload_name):
     def _quote(x):
         return urllib.parse.quote(x.encode("utf-8"))
 
-    request.setHeader(b"Content-Type", media_type.encode("UTF-8"))
+    content_type = (
+        media_type + "; charset=utf-8"
```

Comment:
> The way this is written is it only serves them as UTF-8 by default if there's no other charset given to Synapse at upload time.

it looks like this is unconditionally adding the charset, even if a charset is already given?

Reply:
No. It adds `;charset=utf-8` if and only if the `media_type` value is in `TEXT_CONTENT_TYPES`; so `text/css` is in `TEXT_CONTENT_TYPES` but `text/css; charset=UTF-16` is not. This allows clients to submit custom charsets if they so choose and continue to function.

Reply:
oh right, yes. obvious once you know, but a comment wouldn't go amiss to point out exactly what this code is doing.

Reply:
I gather it's meant to be UTF-8 in capitals, btw.

Reply:
> oh right, yes. obvious once you know, but a comment wouldn't go amiss to point out exactly what this code is doing.

A comment has been added.


> I gather it's meant to be UTF-8 in capitals, btw.

I put this in as well. `@lindalap:matrix.org` told me about the RFCs that list UTF-8 as uppercase.